crates/transport/src/error.rs

```diff
@@ -110,3 +111,65 @@ impl HttpError {
         false
     }
 }
+
+/// Extension trait to implement methods for [`RpcError<TransportErrorKind, E>`].
+pub trait RpcErrorExt {
```

Comment:
why do we need this?

Reply:
Created this to implement `is_retryable_err` and `backoff_hint` methods specifically on `RpcError<TransportErrorKind>` and not the `RpcError<E, ErrResp>` generic type.

Reply:
but this is only implemented for RpcError?

could we at least make this pub(crate)?